controllers/arm.py

```diff
@@ -83,11 +83,19 @@ def __init__(self) -> None:
         self._about_to_run: bool = False
 
     def go_to_setpoint(self, setpoint: Setpoint) -> None:
-        if setpoint != self._target_setpoint:  # and self.at_goal():
+        # If this is a different setpoint, we want to take it
+        # Interrupt what we are doing and move to the new setpoint
+        if setpoint != self._target_setpoint:
             self._about_to_run = True
-            self.next_state("retracting_arm")
-        self._target_setpoint = setpoint
-        self.engage()
+            self._target_setpoint = setpoint
+            self.engage("retracting_arm", force=True)
+
+        # If it is the same setpoint we should check to see if we are actually there
+        # Maybe we got interrupted at some point...
+        # First see if the controller is already running
+        elif not (self.is_executing and self.at_goal()):
```

Comment:
If the arm controller has completed and we continue calling go_to_setpoint I think this will restart the state machine causing it to retract then extend, is the check we want `not executing and not at goal` / `not (executing or at goal)`

Reply:
Correct. Good catch!